### PR TITLE
WIP: (Doc) update doc to match unmodifiable behavior 

### PR DESCRIPTION
**Pull request by MartinWitt**
As discussed in #3770 here are the doc changes.
Still missing is a second run through all interface implementations to fit their sometimes new contract and return an unmodifiable view too.

### CODE REVIEW

File: src/main/java/spoon/pattern/internal/node/ElementNode.java [hunk 1]
```diff
@@ -192,7 +192,9 @@ public boolean replaceNode(RootNode oldNode, RootNode newNode) {
 		}
 		return false;
 	}
-
+	/**
+	 * @return an unmodifiable view of rootNode by metamodel property
```
Comment (MartinWitt) [inline src/main/java/spoon/pattern/internal/node/ElementNode.java:196]: Here I had zero clue what this method really does. Any document proposals?

File: src/main/java/spoon/reflect/declaration/CtModule.java [hunk 2]
```diff
@@ -70,13 +70,17 @@
 
 	@PropertySetter(role = MODULE_DIRECTIVE)
 	<T extends CtModule> T addModuleDirectiveAt(int position, CtModuleDirective moduleDirective);
-
+	/**
+	 * @return an unmodifiable view of module directives
```
Comment (MartinWitt) [inline src/main/java/spoon/reflect/declaration/CtModule.java:74]: The whole class has zero doc? Maybe someone finds time to document all, if he has knowledge about modules.

File: src/main/java/spoon/support/reflect/CtModifierHandler.java [hunk 3]
```diff
@@ -35,7 +35,9 @@ public CtModifierHandler(CtElement element) {
 	public Factory getFactory() {
 		return element.getFactory();
 	}
-
+	/**
+	 * @return an unmodifiable view of extended modifiers
```
Comment (MartinWitt) [inline src/main/java/spoon/support/reflect/CtModifierHandler.java:39]: Here should we describe what "extended" modifiers are. Any proposals?

**Comment by monperrus** [comment]
Per our [contribution process](https://github.com/INRIA/spoon/blob/master/CONTRIBUTING.md), we're closing stale PRs, with no activity in the last 3 months. Should we restart working on this, we'll re-open it.

Thanks!

**Pull request closed without merging**